src/assembler.cpp

```diff
@@ -1188,15 +1188,19 @@ int main(int argc, char* argv[])
                 if (has_swizzle_mask)
                     throw "May not specify a swizzle mask when declaring constants";
 
-                if (ret.GetType() != RegisterType::FloatUniform)
-                    throw "Assigning constants to non-float uniforms is currently unsupported";
 
-                if (values.size() != 4) {
+                if (values.size() != 4 && (ret.GetType() != RegisterType::BoolUniform)) {
                     std::stringstream ss;
                     ss <<"Must specify all register components when assigning constants (expected 4, got " << values.size() << ")";
                     throw ss.str();
                 }
 
+                if (values.size() != 1 && (ret.GetType() == RegisterType::BoolUniform)) {
+                    std::stringstream ss;
+                    ss <<"Must specify a single register component when assigning constants to boolean uniforms (expected 1, got " << values.size() << ")";
+                    throw ss.str();
+                }
+
```

Comment:
Could you add sanity checks for

* making sure integer constants are within the range 0 and 255 inclusive? (additionally, please add a comment saying "TODO: Should negative values be accepted to use LOOP with decreasing counter?"
* making sure boolean constants are either 1 or 0.

?